src/abConfiguration.ts

```diff
@@ -1,7 +1,20 @@
 import TestTrackConfig from './testTrackConfig';
+import Visitor from './visitor';
+import SplitRegistry from './splitRegistry';
+
+export type ABConfigurationOptions = {
+  splitName: string;
+  trueVariant: string | boolean;
+  visitor: Visitor;
+};
 
 class ABConfiguration {
-  constructor(options) {
+  private _splitName: string;
+  private _trueVariant: string | boolean;
+  private _visitor: Visitor;
+  private _splitRegistry: SplitRegistry;
+
+  constructor(options: ABConfigurationOptions) {
     if (!options.splitName) {
       throw new Error('must provide splitName');
```

Comment:
pardon the typescript ignorance... but does typechecking allow us to remove runtime checks like this? i guess i'm asking whether the ts compiler will add runtime checks for when you are calling this from regular old javascript.

Reply:
> i'm asking whether the ts compiler will add runtime checks for when you are calling this from regular old javascript

The compiler won't do that for us. There are some libraries like [io-ts](https://github.com/gcanti/io-ts/blob/HEAD/index.md#the-idea) that allow for us to model types and validate them at runtime.